include/onnxruntime/core/framework/op_kernel_context.h

```diff
@@ -210,7 +210,16 @@ template <>
 inline Tensor* OpKernelContext::Output<Tensor>(int index) {
   OrtValue* p_ml_value = GetOutputMLValue(index);
   ORT_ENFORCE(p_ml_value, "Please fetch output tensor with specified shape.");
+  ORT_ENFORCE(p_ml_value->IsTensor(), "Must be an existing tensor");
```

Comment:
Do we need to add this check now ? Not sure if adding this check in an oft-called method is going to affect the perf of cheap models.

Reply:
You are right, redundant. It is being checked next line

Reply:
similar case in OpKernelContext::Output<SparseTensor>